app/uk/gov/hmrc/gform/models/FieldValueRaw.scala

```diff
@@ -136,11 +136,19 @@ case class FieldValueRaw(
     o = offset.getOrElse(Offset(0))
   } yield Date(f, o, v)
 
-  private lazy val groupOpt: Opt[Group] = fields.fold(groupOptErr)(optGroup)
-  private lazy val groupOptErr: Opt[Group] = InvalidState(s"""Require 'fields' element in Group""").asLeft
-  private def optGroup(rawFields: List[FieldValueRaw]): Opt[Group] = rawFields.map(_.getFieldValue()).partition(_.isRight) match {
-    case (ueorfvs, Nil) => Group(ueorfvs.map(_.right.get)).asRight
-    case (_, ueorfvs) => ueorfvs.map(_.left.get).head.asLeft
+  private lazy val groupOpt: Either[UnexpectedState, Group] = fields.fold(noRawFields)(createGroupFromRawFieldsOrReturnUnexpectedState)
+  private lazy val noRawFields: Either[UnexpectedState, Group] = InvalidState(s"""Require 'fields' element in Group""").asLeft
+  private def createGroupFromRawFieldsOrReturnUnexpectedState(rawFields: List[FieldValueRaw]): Either[UnexpectedState, Group] = {
```

Comment:
The previous names and return types were neat. I don't really understand if we really need to put extra long names and types here?

Let's consider this name for example:
`private def createGroupFromRawFieldsOrReturnUnexpectedState(rawFields: List[FieldValueRaw]): Either[UnexpectedState, Group]`

`create` - we define a function here so it returns/creates something. This word is unnecessary.
`Group` - the function returns a `Group` thus it's obvious that the something we create here is a group thus this word is unnecessary.
`fromRawFields` - the same here: function takes list of raw fields (we can easy spot it reading the type signature)
`OrReturnUnexpectedState` - and the same here, function returns `Opt[Something]` - which means that it returns something or in case of failure `UnexpectedState`. 
In my opinion above words are obvious and don't require extra verbosity. 
Could we please not use such naming?